bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -36,6 +40,12 @@ public void propertyChange(PropertyChangeEvent evt) {
 		MarkerManager mm = v.getSchedule().getMarkerManager();
 		mm.clearMarkers(this, v);
 
+		final TimeZone zone = v.getSchedule().getSite().timezone();
+		final DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("HH:mm") .withZone(zone.toZoneId());
+
+		final Site site = v.getSchedule().getSite();
+		final TwilightBoundedNight tbn = new TwilightBoundedNight(TwilightBoundType.OFFICIAL, v.getSchedule().getMiniModel().getTimestamp(), site);
```

Comment:
I don't understand the model well enough to point you in the right direction but it feels like this should be known. We're making a schedule for a specific night so that should be obtainable from somewhere, I would think.

Reply:
@tpolecat Could you take a look at this PR?  Are the night bounds stored somewhere that we can get to?  What would you suggest?

It seems okay to just calculate them if not but my understanding was that official twilight is too bright and that we used nautical twilight bounds.

Reply:
The timestamp marks when the schedule was constructed so that's not what you want.

A schedule can have multiple blocks, although right now there is exactly one, which is the observing night. So what you probably want to do is examine all the blocks. If there are none then the schedule is totally empty and you should be able to deal with that.

However I can't really figure out what they're asking you to do. What is a "constraint interval"? Is it distinct from a timing window?

Reply:
The qpt is good about warning you when a constraint isn't  met, i.e. when you *can't* schedule an observation.  The idea here was to give the time range when you *can* schedule an observation.